core/designsystem/src/main/java/com/dev/innerview/core/designsystem/component/Dialog.kt

```diff
@@ -0,0 +1,142 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import com.dev.innerview.core.designsystem.theme.InnerViewTheme
+import com.dev.innerview.core.designsystem.theme.Paddings
+
+@Composable
+fun InnerViewDialog(
+    titleText: String = "",
+    contentText: String = "",
+    confirmText: String = "",
+    dismissText: String = "",
+    onDismissRequest: () -> Unit = {},
+    onConfirmRequest: () -> Unit = {},
+    content: (@Composable ColumnScope.() -> Unit)? = null
+) {
+    Dialog(
+        onDismissRequest = { onDismissRequest() }
+    ) {
+        Card(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(Paddings.xlarge),
+            shape = RoundedCornerShape(16.dp),
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .background(MaterialTheme.colorScheme.background)
+                    .padding(Paddings.extra),
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally,
+            ) {
+                Text(
+                    modifier = Modifier.fillMaxWidth(),
+                    text = titleText,
+                    style = MaterialTheme.typography.titleSmall
+                )
+                Spacer(modifier = Modifier.size(16.dp))
+                Text(
+                    modifier = Modifier.fillMaxWidth(),
+                    text = contentText,
+                    style = MaterialTheme.typography.labelMedium.copy(
+                        lineHeight = 18.sp
+                    )
+                )
+                Spacer(modifier = Modifier.size(16.dp))
+                content?.let { it() }
+                Spacer(modifier = Modifier.size(16.dp))
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(horizontal = Paddings.large),
+                    horizontalArrangement = Arrangement.End
+                ) {
+                    Text(
+                        modifier = Modifier.clickable { onDismissRequest() },
+                        text = dismissText,
+                        style = MaterialTheme.typography.labelMedium
+                    )
+                    Text(
+                        modifier = Modifier
+                            .padding(start = 32.dp)
+                            .clickable { onConfirmRequest() },
+                        text = confirmText,
+                        style = MaterialTheme.typography.labelMedium
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO)
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
+private fun InnerViewDialogPreview() {
```

Comment:
🚫 **[ktlint]** <standard:function-naming> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function name should start with a lowercase letter (except factory methods) and use camel case
<!-- __reviewdog__:ChBmYzJmZGMwMGZkY2IwYjUxEgZrdGxpbnQ= -->